### ISSUE TITLE
Flame unusable on Node-DC-EIS workload

### ISSUE THREAD

**Issue by davisjam** [nearform/node-clinic#21]
Ubuntu 16, Node v9.11.1.

1. `~/Desktop/floss/node-clinic/bin.js flame -- node server-cluster.js`
2. `Node-DC-EIS-client $ ./runspec.py -f ./config.json`
3. After client runs (~60 seconds), Ctrl-C
4. Wait for flamegraph

The resulting flamegraph is un-navigable in Firefox.

This is presumably due to [this 0x issue](https://github.com/davidmarkclements/0x/issues/113).

Is there significant benefit to be had from using node clinic -> 0x's flamegraph, or should I use something like [flamebearer](https://github.com/mapbox/flamebearer) which relies on the built-in V8 profiler (as would 0x without the --kernel-tracing)?

In other words, what does 0x's --kernel-tracing mode buy me, other than a hard-to-use flamegraph?

**Comment by mcollina** [comment]
`require('cluster')` cannot be supported in the current set of tooling, but are not documenting that very well right now.
Also, Firefox is not 100% supported in 0x, and we should improve that.

flamebearer copied how 0x sets up the profiler, and they use the same technique by default. 0x also support `--kernel-tracing`, which gives you insights in binary addons and the kernel. The two modes provide significantly different output, and it's worth checking both.

0x also offers some logic on detecting when things are inlined or not as well.

> In other words, what does 0x's --kernel-tracing mode buy me, other than a hard-to-use flamegraph?

If you want to help and send a PR to improve 0x UI performance, you are welcome.

**Comment by davisjam** [comment]
Great, some additional questions:

1. I configured the Node-DC-EIS with nCores 0, so it doesn't actually use the cluster module. Does your code just explode if `require('cluster')` is called, or does it actually need to be used to be problematic? I am assuming the latter and that you don't propagate your instrumentation across instances of the cluster?
2. Firefox support: OK. Is this a problem with the html 0x generates, or just Firefox being slow? I tried with Chrome and performance was better but not great.

**Comment by mcollina** [comment]
The way the data is extracted does not work on child processes.

I think it's a bug in 0x.

**Comment by mcollina** [comment]
Firefox support is now fixed in 0x. Some docs regarding cluster will come out soon.
